### PR TITLE
Support batch update resources

### PR DESCRIPTION
**Pull request by otoyo**
Support batch update resources.

e.g. https://developer.zendesk.com/rest_api/docs/core/tickets#batch-updates

### CODE REVIEW

**Comment by otoyo** [comment]
Failure specs will pass after #345.

File: lib/zendesk_api/actions.rb [hunk 1]
```diff
@@ -276,15 +276,19 @@ def update!(client, attributes = {}, &block)
   module UpdateMany
     # Updates multiple resources using the update_many endpoint.
     # @param [Client] client The {Client} object to be used
-    # @param [Array] ids An array of ids to update
+    # @param [Array] ids_or_attributes_array An array of ids or arributes including ids to update
     # @param [Hash] attributes The attributes to update resources with
     # @return [JobStatus] the {JobStatus} instance for this destroy job
-    def update_many!(client, ids, attributes)
+    def update_many!(client, ids_or_attributes_array, attributes = {})
```
Comment (otoyo) [inline lib/zendesk_api/actions.rb]: `ids_or_attributes_array` naming looks not good to me...

**Comment by otoyo** [comment]
@grosser Would you be able to review this also, please?

File: lib/zendesk_api/actions.rb (hunk 1 above)
Comment (grosser) [reply lib/zendesk_api/actions.rb]: I'd leave off the _array ... and make `attributes=nil` to mke the check simpler

**Comment by grosser** [comment]
let me know what you think of my suggestions ... looks good to merge either way

File: lib/zendesk_api/actions.rb (hunk 1 above)
Comment (otoyo) [reply lib/zendesk_api/actions.rb]: Sounds good to me but L283 might need conditions.

**Comment by otoyo** [comment]
I've left `_array`.

File: spec/core/bulk_actions_spec.rb [hunk 2]
```diff
@@ -47,22 +47,44 @@
     subject { ZendeskAPI::BulkTestResource }
 
     context "update_many!" do
-      let(:attributes) { { :name => 'A', :age => 25 } }
+      context "arity: 3" do
```
Comment (grosser) [inline spec/core/bulk_actions_spec.rb]: `updating a list of ids`

File: spec/core/bulk_actions_spec.rb [hunk 3]
```diff
@@ -47,22 +47,44 @@
...
+        it 'returns a JobStatus' do
+          expect(@response).to be_instance_of(ZendeskAPI::JobStatus)
+          expect(@response.id).to eq('ghi')
+        end
       end
 
-      it 'returns a JobStatus' do
-        expect(@response).to be_instance_of(ZendeskAPI::JobStatus)
-        expect(@response.id).to eq('ghi')
+      context "arity: 2" do
```
Comment (grosser) [inline spec/core/bulk_actions_spec.rb]: `updating with multiple attribute hashes`

**Comment by grosser** [comment]
1.15.0 :tada:

**Comment by otoyo** [comment]
Thank you for your review!

**Comment by grosser** [comment]
thx for some great PRs :)

**Pull request merged**